### ISSUE TITLE
google_compute_disk created from snapshot, forces new resource then apply once more

### ISSUE THREAD

**Issue by z1nkum** [terraform-providers/terraform-provider-google#218]
Looks like google_compute_disk was broken since v0.9.6 (or v0.9.7)

Create disk snapshot and then disk from this snapshot (clone disk operation)
If we execute `terraform apply` once more newly created disk will be destroyed and recreated

### Terraform Version
Terraform v0.9.11

### Affected Resource(s)
google_compute_disk

### Terraform Configuration Files
```hcl
resource "google_compute_snapshot" "db-shard1-4-root-snapshot" {
  name  = "db-shard1-4-root-snapshot"
  source_disk  = "db-shard1-4"
  zone  = "${google_compute_instance.db-shard1-4.zone}"
}

resource "google_compute_disk" "db-shard1-4-root-disk-from-snapshot" {
  name  = "db-shard1-4-root-cloned-disk-${count.index + 1}"
  type  = "pd-standard"
  zone  = "${var.zone-chooser[count.index]}"
  snapshot = "${google_compute_snapshot.db-shard1-4-root-snapshot.name}"
  count = 1
}
```

### Debug Output
```
-/+ google_compute_disk.db-shard1-4-root-disk-from-snapshot
    disk_encryption_key_sha256: "" => "<computed>"
    name:                       "db-shard1-4-root-cloned-disk-1" => "db-shard1-4-root-cloned-disk-1"
    self_link:                  "https://www.googleapis.com/compute/v1/projects/projectnamehere/zones/us-east1-d/disks/db-shard1-4-root-cloned-disk-1" => "<computed>"
    size:                       "10" => "0"
    snapshot:                   "https://www.googleapis.com/compute/v1/projects/projectnamehere/global/snapshots/db-shard1-4-root-snapshot" => "db-shard1-4-root-snapshot" (forces new resource)
    type:                       "pd-standard" => "pd-standard"
    users.#:                    "0" => "<computed>"
    zone:                       "us-east1-d" => "us-east1-d"


Plan: 1 to add, 0 to change, 1 to destroy.
```

### Expected Behavior
No changes expected on 2nd and others "terraform apply"

### Actual Behavior
Terraform decide to recreate disks (why??)

### Steps to Reproduce
List the steps required to reproduce the issue:
Execute `terraform apply`
And once more: `terraform apply`

**Comment by nodesocket** [comment]
See https://github.com/terraform-providers/terraform-provider-google/issues/112. This is fixed in `v0.10.0`. See the following new syntax and example:

````
resource "google_compute_disk" "blog" {
    name = "blog"
    type = "pd-ssd"
    zone = "${var.blog_zone}"
    snapshot = "blog-2017-07-12-1"
    size = 10
}

resource "google_compute_instance" "blog" {
    ...
    boot_disk {
        source = "${google_compute_disk.blog.name}"
        auto_delete = true
    }
    ...
}
````

**Comment by nodesocket** [comment]
Maybe I spoke too soon. This seems to actually still be a bug.

```
iMac ➜  terraform git:(master) ✗ terraform apply
Error applying plan:

1 error(s) occurred:

* google_compute_disk.blog (destroy): 1 error(s) occurred:

* google_compute_disk.blog: Error detaching disk persistent-disk-0 from instance elastic-byte/https://www.googleapis.com/compute/v1/projects/my-project/zones/us-west1-a/blog: googleapi: Error 400: Invalid value 'https://www.googleapis.com/compute/v1/projects/my-project/zones/us-west1-a'. Values must match the following regular expression: '[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?', invalidParameter

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

**Comment by rileykarson** [comment]
@nodesocket: It looks like the snapshot forcing a recreate is the same issue as this, but the detach failing looks like a separate issue to me; would you mind filing another issue with the first block & with a repro config?

**Comment by nodesocket** [comment]
@rileykarson there is already an issue that was ported over: https://github.com/terraform-providers/terraform-provider-google/issues/112

**Comment by rileykarson** [comment]
Oh cool - I'd missed that one, thanks!

**Comment by danawillow** [comment]
This was fixed in #238, closing.
